src/detect/azureDevOpsPipelines.ts

```diff
@@ -89,6 +89,11 @@ export class AzureDevOpsPipelinesDetector extends BaseDetector {
       }
     }
 
+    if (!targetRef && this.shouldDetectTargetType('commit')) {
```

Comment:
Seems like `!targetRef` is redundant, but doesn't hurt to keep :)
```suggestion
    if (this.shouldDetectTargetType('commit')) {
```

Reply:
I think we need it. Otherwise it overwrites the targetRef that is found when looking for a PR environment.

Reply:
Each if statement excludes the other, so if `this.shouldDetectTargetType('commit')` is true, `this.shouldDetectTargetType('pull-request') || this.shouldDetectTargetType('merge-request')` will always be false, right? I may be missing something 🤔 . Anyway, it's okay to keep :)

Reply:
By default compost detects both PRs and commits unless it's limited it with the `--target-type` flag.  So both can be true if there's no flag to limit the detection.

Reply:
Ah, got it! Thank you for the explanation 👍!